packages/desktop-client/src/components/settings/Tags.tsx

```diff
@@ -0,0 +1,233 @@
+    );
+  } else {
+    return (
+      <Button
+        variant="bare"
+        className={getTagCSS(tag)}
+        onPress={() => onTrashTag(tag)}
+      >
+        {formattedTag}
+        &nbsp;
+        {tag !== '*' ? (
+          <SvgClose width={10} height={10} />
+        ) : (
+          <SvgRefresh width={10} height={10} />
+        )}
+      </Button>
+    );
+  }
+};
+
+export function TagsSettings() {
+  const sidebar = useSidebar();
+  const [tags, setTagsPref] = useTags();
+  const getTagCSS = useTagCSS();
+  const [newTag, setNewTag] = useState('');
+  const [newColor, setNewColor] = useState(purple700);
+  const [errorMsg, setErrorMsg] = useState('');
+  const [trashMode, setTrashMode] = useState(false);
+
+  const onAddTagColor = () => {
+    if (!newTag.trim()) {
+      setErrorMsg(t('Tag name cannot be empty'));
+      return;
+    }
+
+    if (!newColor.match(/^#(?:[0-9a-f]{3}){1,2}$/i)) {
+      setErrorMsg(t('Invalid HEX color'));
+      return;
+    }
+
+    setErrorMsg('');
+    setTagsPref({ ...tags, [newTag.trim()]: newColor });
+    setNewTag('');
+  };
+
+  const actionButtonStyle = {
+    borderWidth: 0,
+    backgroundColor: 'transparent',
+    marginLeft: 'auto',
+  };
+
+  return (
+    <Setting
+      primaryAction={
+        <View
+          style={{
+            flexDirection: 'column',
+            gap: '1em',
+            width: '100%',
+            [`@media (min-width: ${
+              sidebar.floating
+                ? tokens.breakpoint_small
+                : tokens.breakpoint_medium
+            })`]: {
+              flexDirection: 'column',
+            },
+          }}
+        >
+          {errorMsg && (
+            <FormError
+              style={{
+                paddingTop: 5,
+                marginLeft: styles.mobileEditingPadding,
+                marginRight: styles.mobileEditingPadding,
+                color: theme.errorText,
+              }}
+            >
+              * {errorMsg}
+            </FormError>
+          )}
+          <Form
+            style={{
+              display: 'flex',
+              flexDirection: 'row',
+              gap: '1em',
+            }}
+            onSubmit={e => {
+              e.preventDefault();
+              onAddTagColor();
+            }}
+          >
+            <Input
+              id="tag-field"
+              placeholder="tag"
+              value={newTag}
+              onChangeValue={setNewTag}
+            />
+            <View
+              style={{
+                display: 'flex',
+                flexDirection: 'row',
+                gap: '8px',
+              }}
+            >
+              <View style={{ margin: 'auto 0' }}>
+                <ColorPicker
+                  value={newColor}
+                  onChange={color => setNewColor(color.toString('hex'))}
+                >
+                  <Button
+                    variant="bare"
+                    className={css(getTagCSS('', newColor))}
```

Comment:
You don't need css here since getCSS already returns a className
```suggestion
                    className={getTagCSS('', newColor)}
```